src/exoticatechnologies/util/StacktraceUtils.kt

```diff
@@ -0,0 +1,40 @@
+package exoticatechnologies.util
+
+object StacktraceUtils {
+
+    /**
+     * Method for unwinding the stacktrace and returning a string containing a whole "typical" stacktrace
+     * you see in logs. Whether the imminent place of happening will be at the top or the bottom of the trace
+     * depends on the [startToEnd] argument, which defaults to [true] (*start* being the place of happening and *end*
+     * being the place it originated from)
+     *
+     * @param stacktraceArray array containing [StackTraceElement]s
+     * @param startToEnd whether the place you should start looking from will be located at the top or the bottom of the stacktrace
+     * @return printable, formatted stacktrace, as a String
+     */
+    fun unwindStacktrace(stacktraceArray: Array<StackTraceElement?>, startToEnd: Boolean = true): String {
```

Comment:
too bad this doesn't actually print out a stacktrace identical to the actual one you get when something crashes.

the problem is that it only 'knows' about `StackTraceElement[]` and not the actual exception itself which is missing in it's printout. Consider refactoring, or offering a version which just takes an `Exception` then ends up calling this one internally or something.

Reply:
new default parameter makes it print out a stacktrace identical to the 'normal' one, minus the Exception's name. the new method ~~prints~~ stringifies out the exception's name followed by the stacktrace.